### PR TITLE
fix: update README and add uploader field to the create and update record functions

### PR DESCRIPTION
**Pull request by mjmartinson**
Update README and add uploader field to the create and update record functions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,30 +74,164 @@ An example of one such record:
...
 
 These queries are handled via requests and wrapped into the index client.
 
-### Create a record
 
-***TODO***
+  ### Create a record ###
+
+
+# Method: create
```
Comment (Avantol13) [inline README.md]: you should use lower level header here. `####`, don't reuse the top-level `#`

File: README.md [hunk 2]
```diff
@@ -74,30 +74,164 @@ An example of one such record:
...
 All queries to the index are made through HTTP using JSON data payloads.
 This gives a simple means of interaction that is easily accessible to any
 number of languages.
 
 These queries are handled via requests and wrapped into the index client.
 
-### Create a record
 
-***TODO***
+  ### Create a record ###
```
Comment (Avantol13) [inline README.md]: you don't technically need the trailing `###`

File: README.md [hunk 3]
```diff
@@ -74,30 +74,164 @@ An example of one such record:
...
+  baseid (str): optional baseid to group with previous entries versions
+  version (str): entry version string
+
+Return Value:
+  Document: indexclient representation of an entry in indexd
+
+Example: 
+
+ indexclient.create(
+    hashes = {'md5': ab167e49d25b488939b1ede42752458b'},
```
Comment (Avantol13) [inline README.md]: your example should just show passing the `did` instead of hash and size

File: README.md [hunk 4]
```diff
@@ -74,30 +74,164 @@ An example of one such record:
...
 
-### Create a record
 
-***TODO***
+  ### Create a record ###
+
+
+# Method: create
+
+Required Arguments:
```
Comment (Avantol13) [inline README.md]: i actually would not include any of this or optional args since the code may be changed and we shouldn't have to update this too. So honestly I'd just have the `Example` without this.

File: README.md [hunk 5]
```diff
@@ -74,30 +74,164 @@ An example of one such record:
...
+  ### Retrieve a record ###
+
+
+# Method: get
+
+Required Argument:
+  did (str): GUID (Global Unique Identifier) for the new indexd to be made
+
+Return Value:
+  Document: indexclient representation of an entry in indexd
```
Comment (Avantol13) [inline README.md]: like I said above I would not include the required or return blocks. I would use actual values for the `did` arg though. like show `indexclient.get(did="dg.1234/03eed607-acb0-4532-b0ee-9e3766b1aa6e")

File: README.md [hunk 6]
```diff
@@ -74,30 +74,164 @@ An example of one such record:
...
+
+Example:
+
+ indexclient.get(did)
```
Comment (Avantol13) [inline README.md]: you should surround this in a code block with ` ` ` 
```
like this
```

File: README.md [hunk 7]
```diff
@@ -74,30 +74,164 @@ An example of one such record:
...
+Required Argument:
+  dids (list): dids (list): list of dids for potential documents
+
+Return Value:
+  list: Document objects representing index records
+
+Example:
+
+ dids = [
+   'did1',
```
Comment (Avantol13) [inline README.md]: you should use actual GUIDs here as an example

File: README.md [hunk 8]
```diff
@@ -74,30 +74,164 @@ An example of one such record:
...
 
-### Name a record
+Lastly: Update all the local changes that were made to indexd using the 
+        Document patch method: doc.patch()
 
-***TODO***
+Example:
 
-### Retrieve a record
+ 1. doc = indexclient.get('did') # or any other get method (global_get,
```
Comment (Avantol13) [inline README.md]: just use a code block instead of 1.2.3

File: indexclient/client.py [hunk 9]
```diff
@@ -18,6 +18,7 @@
     "acl",
     "authz",
     "urls_metadata",
+    "uploader",
```
Comment (Avantol13) [inline indexclient/client.py]: why was this change necessary?

Comment (mjmartinson) [reply indexclient/client.py]: I guess it is not completely necessary, but it is part of the record schema. If you want to leave it out I will just remove.

Comment (Avantol13) [reply indexclient/client.py]: I would say it's safer to leave it out for now

File: README.md [hunk 10]
```diff
@@ -74,30 +74,123 @@ An example of one such record:
...
 
 These queries are handled via requests and wrapped into the index client.
 
-### Create a record
 
-***TODO***
+  ### Create a record 
+
+
+### Method: create
```
Comment (Avantol13) [inline README.md]: I would make this `####` so it's under "create a record" (same with other methods below)

File: README.md [hunk 11]
```diff
@@ -74,30 +74,123 @@ An example of one such record:
...
-***TODO***
+Lastly: Update all the local changes that were made to indexd using the 
+        Document patch method: doc.patch()
 
-### Delete a record
+Example:
 
-***TODO***
+```python
+		doc = indexclient.get(did="dg.1234/03eed607-acb0-4532-b0ee-9e3766b1aa6e"')
```
Comment (Avantol13) [inline README.md]: you can just remove all the leading spaces/tabs inside the python blocks (same for below)
